### ISSUE TITLE
Lab 04 - RenderPlot

### ISSUE THREAD

**Issue by ecking** [DS4PS/cpp-526-spr-2020#19]
Hi there, 

I'm not entirely sure whats going on here. I placed my code from last week into the renderPlot function but I'm getting this error: 

`Error in renderPlot({ : could not find function "renderPlot"`

I've googled it and still not sure why I'm receiving this. The packages are here and updated. I've restarted R Studio. But alas, I'm still stuck.

**Comment by lecy** [comment]
That's a great question. Since you include the shiny runtime argument in your YAML header: 

```r
---
title: "Widget Demo"
runtime: shiny
output: html_document
---
```

You never actually load the shiny package. If you want to test the code directly, since **renderPlot()** belongs to the shiny package you need to load it first:

```r
library( shiny )
```

Be forewarned, though, the output is a bunch of javascript so it will not make any sense. You are better-off running your file to see how your current code renders: 

![image](https://user-images.githubusercontent.com/1209099/74114168-b34a6a00-4b65-11ea-868a-27682b6d8752.png)

**Comment by ecking** [comment]
Hmmm. Am I supposed to see my graph after running this?

**Comment by lecy** [comment]
Yes. I can't tell what you are doing, though, without code.

**Comment by lecy** [comment]
It should look something like this: 

![cpp-526-lab-04-demo](https://user-images.githubusercontent.com/1209099/74123099-bad03a00-4b8a-11ea-810d-726457925fcb.gif)

**Comment by ecking** [comment]
Hmm I think I'm completely lost. I took a look at the demo rmd doc and I'm still not 100% on what to do. I'm looking at the example and then I hit run... is that supposed to open this app? Or am I supposed to do something additional with this demo file? 

I do see, I can go to the shiny web app  and running that example opens the above app. But I'm not entirely sure what to do.

**Comment by ecking** [comment]
WOW nevermind... would help if i hit "Run document" *face palm* :-D I'm on track for the time being!

**Comment by lecy** [comment]
Have you tried turning it off and on again? Maybe jiggle the handle? :-)

**Comment by msatre-png** [comment]
I hate to say it but I'm having the same issue.  I paste my Week 3 code in the code chunk with the renderPlot() function and when I run it I don't get a error but I also don't see my graph.  I just get the 
<shiny.render.function> at the botttom of the below paste.  I think that I've installed Shiny properly and loaded it but must be doing something wrong or there is another step I'm not getting (even with the help from the previous posts on this topic).  My code:


```{r load_data}
install.packages( "shiny" )
install.packages( "flexdashboard" )
library( Lahman )
data( Teams )
library(pander)
library(dplyr)
library(shiny)
library(flexdashboard)
```

```{r global, include=FALSE}
library( dplyr )
library( Lahman )
data( Teams )

ave.so <- Teams$SO / Teams$G  # average strike-outs per game each season 
year <- Teams$yearID

# min and max values needed to create your plot window 
ave.so.min <- min( ave.so, na.rm=T )
ave.so.max <- max( ave.so, na.rm=T )

# the league average for each season
league.ave <- tapply( X=ave.so, INDEX=as.factor(year), FUN="mean", na.rm=T )
league.year <- as.numeric( names( league.ave ) )
```


```{r}

renderPlot({
  
I copy and pasted my Week 3 Code here

})

```

<shiny.render.function> 
-----------------------------------------------------------------------

**Comment by lecy** [comment]
@msatre-png Have you tried the "Run Document" option at the top of the page? When using runtime shiny this option replaced your "knit" option. 

![image](https://user-images.githubusercontent.com/1209099/74202228-1cdf7c80-4c29-11ea-9b0f-c4421a1a20e4.png)

**Comment by ecking** [comment]
@msatre-png Yeah to the left, there is a run document button. Don't worry, I didn't see it either LOL. When you click that, a new window will open and look like it's supposed to.

**Comment by msatre-png** [comment]
I'm embarrassed, I didn't see it even with the heads up you both gave me.
However, when I do Run the Document I'm getting the following error?
ERROR: Trying to use CRAN without setting a mirror

**Comment by ecking** [comment]
@msatre-png that one I can't help out with. But don't worry I spent longer than I should have googling and asking people, just to come find out that all I was missing was a button LOL. 

and later I spent longer that needed hung up on something that was as simple as changing 'names' to "name"... so cool. really rocking it on this assignment.

**Comment by lecy** [comment]
You are trying to install packages during a knit (or in this case a run), which is bad. 

Install packages once, then comment them out (I like to keep the code in case I'm on a different computer and need to re-install the packages). But do not try to run it during a knit - it can cause package conflicts and create problems. 

```r
# install.packages( "shiny" )
# install.packages( "flexdashboard" )
library( Lahman )
data( Teams )
library(pander)
library(dplyr)
library(shiny)
library(flexdashboard)
```

**Comment by msatre-png** [comment]
Thank you both for steering me in the right direction and the encouragement.  Certainly not my best lab thus far.  But I'm getting there:)

**Comment by ecking** [comment]
Not my best either! Don’t worry!
